### PR TITLE
Provide a few default args for numpy translation

### PR DESCRIPTION
**Pull request by umanwizard**
Add automatic translations for a few argument names that commonly differ between PyTorch and NumPy.

For now, they are as follows:

* `keepdim` -> `keepdims`
* `dim` -> `axis`
* `input` -> (any of `a`, `x`, `x1`)
* `other` -> `x2`

Basic examples:
```python
>>> t=torch.randn(10,10)
>>> torch.sum(x=t, axis=1)
tensor([ 0.5199, -0.3768,  4.3619, -0.9105,  1.1804,  1.0837, -0.9036,  0.2365,
         1.1171, -0.0999])
```
```python
>>> torch.add(x1=5, x2=6)
tensor(11)
```

The additional overhead is zero when using traditional PyTorch argument names, and a few (usually 1) extra PyDict lookups when using NumPy argument names.

### CODE REVIEW

**Comment by umanwizard** [comment]
Related to #2228

File: torch/csrc/utils/python_arg_parser.cpp [hunk 1]
```diff
@@ -32,6 +32,28 @@ static std::unordered_map<std::string, ParameterType> type_map = {
...
+// In particular, NumPy sometimes uses "x" and sometimes "a" for the main input tensor.
+// Rather than annotate each function separately with whether it should take "x" or "a",
+// just try both.
+//
+// TODO: Allow individual functions to specify non-default translations:
+// For example, `torch.pow` should translate "exponent" to "x2".
+static const std::unordered_map<std::string, std::vector<std::string>> numpy_compatibility_arg_names = {
+  {"dim", {"axis"}},
+  {"keepdim", {"keepdims"}},
+  {"input", {"x", "a", "x1"}},
```
Comment (colesbury) [inline torch/csrc/utils/python_arg_parser.cpp]: I'm a bit ambivalent about the `input` and `other` aliases. I'm not sure people use them as kwargs often enough to make it worthwhile. There's also a long tail of names that NumPy uses instead of "input", including  "A", "arr", "ary", "m", "a1", "array".

**Review by colesbury — APPROVED** [review]
LGTM

As I wrote inline, I'm ambivalent about the aliases for "input" and "other", but I'm not opposed to them.

File: torch/csrc/utils/python_arg_parser.cpp (hunk 1 above)
Comment (umanwizard) [reply torch/csrc/utils/python_arg_parser.cpp]: > I'm a bit ambivalent...

I don't know how often people use them as kwargs. But I want to be as complete as possible.

> There's also a long tail...

Yes, and I've also seen `p`, and who knows what else there is. I think we need to decide case-by-case, depending on how common they are, whether to include them in the default common args here, or annotate individual functions with alternate arg names (I will introduce a facility for doing that in a future PR).
